src/xercesc/internal/ReaderMgr.hpp

```diff
@@ -208,36 +214,96 @@ private :
     ReaderMgr(const ReaderMgr&);
     ReaderMgr& operator=(const ReaderMgr&);
 
+    // -----------------------------------------------------------------------
+    //  Private data types
+    // -----------------------------------------------------------------------
+    class ReaderData : public XMemory
+    {
+    public  :
+      // ---------------------------------------------------------------------
+      //  Constructors and Destructor
+      // ---------------------------------------------------------------------
+      ReaderData
+      (    XMLReader* const     reader
+         , XMLEntityDecl* const entity
+         , const bool           adoptEntity
+      );
+
+      ~ReaderData();
+
+      // ----------------------------------------------------------------------
+      //  Getter methods
+      // ----------------------------------------------------------------------
+      XMLReader* getReader() const;
+      XMLEntityDecl* getEntity() const;
+      bool getEntityAdopted() const;
+
+      XMLEntityDecl* releaseEntity();
+
+    private :
+      // ---------------------------------------------------------------------
+      //  Unimplemented constructors and operators
+      // ---------------------------------------------------------------------
+      ReaderData();
+      ReaderData(const ReaderData&);
+      ReaderData& operator=(const ReaderData&);
+
+      // ---------------------------------------------------------------------
+      //  Private data members
+      //
+      //  fReader
+      //      This is the pointer to the reader object that must be destroyed
+      //      when this object is destroyed.
+      //
+      //  fEntity
+      //  fEntityAdopted
+      //      This is the pointer to the entity object that, if adopted, must
+      //      be destroyed when this object is destroyed.
+      //
+      //      Note that we need to keep up with which of the pushed readers
+      //      are pushed entity values that are being spooled. This is done
+      //      to avoid the problem of recursive definitions.
+      // ---------------------------------------------------------------------
+      XMLReader*     fReader;
+      XMLEntityDecl* fEntity;
+      bool           fEntityAdopted;
```

Comment:
```suggestion
        // ---------------------------------------------------------------------
        //  Unimplemented constructors and operators
        // ---------------------------------------------------------------------
        ReaderData();
        ReaderData(const ReaderData&);
        ReaderData& operator=(const ReaderData&);

        // ---------------------------------------------------------------------
        //  Private data members
        //
        //  Reader
        //      This is the pointer to the reader object that must be destroyed
        //      when this object is destroyed.
        //
        //  fEntity
        //  fEntityAdopted
        //      This is the pointer to the entity object that, if adopted, must
        //      be destroyed when this object is destroyed.
        //
        //      Note that we need to keep up with which of the pushed readers
        //      are pushed entity values that are being spooled. This is done
        //      to avoid the problem of recursive definitions.
        // ---------------------------------------------------------------------
        XMLReader*     Reader;
        XMLEntityDecl* fEntity;
        bool           fEntityAdopted;
```